### PR TITLE
baremetal: fix handling of the "fields" query argument

### PR DESCRIPTION
**Pull request by dtantsur**
Ironic expects a comma-separated string. Add a new feature to
BuildQueryString to choose the format of slices.

Closes: #2960

### CODE REVIEW

**Review by mandre — COMMENTED** [review]
LGTM. This is a rather elegant solution and I quite like it.

**Review by mandre — DISMISSED** [review]

**Comment by dtantsur** [comment]
> @[dtantsur](https://github.com/dtantsur) dtantsur dismissed [mandre](https://github.com/mandre)’s [stale review](https://github.com/gophercloud/gophercloud/pull/2961#pullrequestreview-1924881263)

To be clear, I obvious did NOT do that :-/ (I miss gerrit so much)

**Comment by mandre** [comment]
> > @[dtantsur](https://github.com/dtantsur) dtantsur dismissed [mandre](https://github.com/mandre)’s [stale review](https://github.com/gophercloud/gophercloud/pull/2961#pullrequestreview-1924881263)
> 
> To be clear, I obvious did NOT do that :-/ (I miss gerrit so much)

Let me try again, otherwise you might need a rebase after https://github.com/gophercloud/gophercloud/pull/2964 merged.

**Review by mandre — DISMISSED** [review]

**Review by mandre — APPROVED** [review]
